source/MaterialXGenOgsXml/OgsXmlGenerator.cpp

```diff
@@ -0,0 +1,243 @@
+//
+// TM & (c) 2017 Lucasfilm Entertainment Company Ltd. and Lucasfilm Ltd.
+// All rights reserved.  See LICENSE.txt for license.
+//
+
+#include <MaterialXGenOgsXml/OgsXmlGenerator.h>
+#include <MaterialXGenOgsXml/GlslFragmentGenerator.h>
+
+#include <MaterialXGenShader/Shader.h>
+
+#include <MaterialXFormat/XmlIo.h>
+#include <MaterialXFormat/PugiXML/pugixml.hpp>
+
+using namespace pugi;
+
+namespace MaterialX
+{
+
+namespace
+{
+    // Data types used by OGS
+    static const std::unordered_map<const TypeDesc*, const pugi::char_t*> OGS_TYPE_MAP =
+    {
+        { Type::BOOLEAN, "bool" },
+        { Type::FLOAT, "float" },
+        { Type::INTEGER, "int" },
+        { Type::STRING, "int" },
+        { Type::COLOR2, "float2" },
+        { Type::COLOR3, "float3" },
+        { Type::COLOR4, "float4" },
+        { Type::VECTOR2, "float2" },
+        { Type::VECTOR3, "float3" },
+        { Type::VECTOR4, "float4" },
+        { Type::MATRIX33, "float4x4" },
```

Comment:
Does this mean we promote matrix33 to be matrix44 in code or just the exposed uniforms. I assume it can't be the latter? Not sure..